### PR TITLE
Allow setting sshkey and kernel params for debugging IPA

### PR DESCRIPTION
**Pull request by derekhiggins**

### CODE REVIEW

**Comment by derekhiggins** [comment]
The previous PR to do this is here https://github.com/metal3-io/ironic-image/pull/150
but it went stale and was before we moved to jinja for the config

**Comment by derekhiggins** [comment]
cc @maelk , includes a proposal to remove console=ttyS0 we had hardcoded in, and use IRONIC_KERNEL_PARAMS instead

**Comment by elfosardo** [comment]
hey @derekhiggins are this method or any of the parameters documented anywhere?

**Comment by derekhiggins** [comment]
> hey @derekhiggins are this method or any of the parameters documented anywhere?

Nope, I'll add some. thanks

**Comment by maelk** [comment]
Would you mind creating a PR in BMO to maintain the current behaviour when merging this change ? There's two deployment methods in BMO, through ironic-deployment folder and with tools/run_local_ironic.sh. both refer to a configmap that need to be updated . There is also one in metal3-dev-env, in the 03_* script. Would you be ok modifying those ? thank you

**Comment by derekhiggins** [comment]
> Would you mind creating a PR in BMO to maintain the current behaviour when merging this change ? There's two deployment methods in BMO, through ironic-deployment folder and with tools/run_local_ironic.sh. both refer to a configmap that need to be updated . There is also one in metal3-dev-env, in the 03_* script. Would you be ok modifying those ? thank you

Yup, my intention is that it would only be maintained for virtual environments, I'll update those

File: runhttpd.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ else
 fi
 
 # Copy files to shared mount
-cp /tmp/inspector.ipxe /shared/html/inspector.ipxe
+render_j2_config /tmp/inspector.ipxe.j2 /shared/html/inspector.ipxe
```
Comment (dtantsur) [inline runhttpd.sh:29]: Handling of options below should be removed.

Comment (derekhiggins) [reply runhttpd.sh:29]: Can you elaborate on this, I'm not sure what you mean?

Comment (dtantsur) [reply runhttpd.sh:29]: Unfortunately, github does not allow me to comment on the code a bit below. It has the remaining bits of the old approach to templating, they should be removed.

Comment (derekhiggins) [reply runhttpd.sh:29]: >  cp /tmp/dualboot.ipxe /shared/html/dualboot.ipxe
>  cp /tmp/uefi_esp.img /shared/html/uefi_esp.img

these line? Those aren't templates

Comment (derekhiggins) [reply runhttpd.sh:29]: > > cp /tmp/dualboot.ipxe /shared/html/dualboot.ipxe
> > cp /tmp/uefi_esp.img /shared/html/uefi_esp.img
> 
> these lines? Those aren't templates

**Comment by stbenjam** [comment]
Needs rebase.  https://github.com/metal3-io/baremetal-operator/pull/736 should land soon. Anything else we need to keep current behaviour?

**Comment by derekhiggins** [comment]
> Needs rebase. [metal3-io/baremetal-operator#736](https://github.com/metal3-io/baremetal-operator/pull/736) should land soon. Anything else we need to keep current behaviour?

rebase done, I'm not aware of anything else yet to merge to keep current behaviour.

**Review by maelk — COMMENTED** [review]
/lgtm

**Pull request merged**